### PR TITLE
Add observability-bundle

### PR DESCRIPTION
**Pull request by mcbenjemaa**
towards https://github.com/giantswarm/giantswarm/issues/24246

### CODE REVIEW

**Comment by erkanerol** [comment]
@mcbenjemaa Can you add more details into the PR description?

As far as I understand, we will install these apps to MC for each WC.
- prometheus-operator-crd
- prometheus-operator-app
- prometheus-agent

How does this work with prometheus-meta-operator? Will this increase resources consumption of MC per WC?

**Comment by QuentinBisson** [comment]
@erkanerol we will be deploying those apps in the WC themselves :)

In the MC we will have:

- prometheus-meta-operator
- prometheus-operator-app
- prometheus-operator-crd
- prometheus-agent-app

In the WC we will have:

- prometheus-operator-app
- prometheus-operator-crd
- prometheus-agent-app

**Comment by erkanerol** [comment]
@QuentinBisson We use `default-apps-openstack` for MCs too. Therefore, the apps here shouldn't conflict with the ones in https://github.com/giantswarm/openstack-app-collection/tree/master/flux-manifests. As far as I see, we will have some conflicts.

**Comment by mcbenjemaa** [comment]
@erkanerol the observability-bundle installs:

- prometheus-operator-app
- prometheus-operator-crd
- prometheus-agent-app

to each WC. So no conflicts in the MC. 

the apps on MC, are installed via flux.

**Comment by QuentinBisson** [comment]
@erkanerol we will have conflicts and solve them in time :D

**Comment by erkanerol** [comment]
Note that this PR introduces these footprints in each WC:

```
prometheus-operator-app-operator:
    resources:
      limits:
        cpu: 200m
        memory: 200Mi
      requests:
        cpu: 100m
        memory: 100Mi

prometheus-prometheus-agent-0:
    resources:
      limits:
        cpu: 100m
        memory: 50Mi
      requests:
        cpu: 100m
        memory: 50Mi
```

**Comment by QuentinBisson** [comment]
Sure think :) Is that an issue?

In a next release, prometheus-operator will run with VPA

**Review by QuentinBisson — APPROVED** [review]

File: helm/default-apps-openstack/templates/apps.yaml [hunk 1]
```diff
@@ -8,17 +8,27 @@ metadata:
     chart-operator.giantswarm.io/force-helm-upgrade: "{{ .forceUpgrade }}"
   labels:
     {{- include "labels.common" $ | nindent 4 }}
+    {{- if .inCluster }}
+    app-operator.giantswarm.io/version: 0.0.0
+    {{- end }}
```
Comment (kopiczko) [inline helm/default-apps-openstack/templates/apps.yaml:13]: Why do we need this `inCluster` thing and when is it supposed to be used?

Comment (kopiczko) [reply helm/default-apps-openstack/templates/apps.yaml:13]: Ah got it. This is always installed in the MC. Mind bending.

**Comment by mcbenjemaa** [comment]
@erkanerol https://github.com/giantswarm/app/pull/266

File: helm/default-apps-openstack/templates/apps.yaml (hunk 1 above)
Comment (mcbenjemaa) [reply helm/default-apps-openstack/templates/apps.yaml:13]: same as the default-apps, it should be handled by the app-operator of MC.

**Review by pipo02mix — APPROVED** [review]

**Comment by mcbenjemaa** [comment]
@erkanerol the app-admission fix was released

File: helm/default-apps-openstack/values.schema.json [hunk 2]
```diff
@@ -424,10 +471,5 @@
                 }
             }
         }
-    },
-    "required": [
```
Comment (erkanerol) [inline helm/default-apps-openstack/values.schema.json:428]: I think this `required` section is deleted accidentally. @mcbenjemaa @QuentinBisson

**Pull request merged**